src/webnn_native/openvino/GraphIE.cpp

```diff
@@ -454,6 +455,55 @@ namespace webnn_native { namespace ie {
         return {};
     }
 
+    MaybeError Graph::AddSlice(const op::Slice* slice) {
+        auto input = mGraphNodeMap[slice->Inputs()[0].Get()];
+        dimensions_t inputShape;
+        ngraph_get_shape(input, &inputShape);
+        std::vector<int32_t> beginVec;
+        std::vector<int32_t> endVec;
+        std::vector<int32_t> starts = slice->GetStarts();
+        std::vector<int32_t> sizes = slice->GetSizes();
+        std::vector<int32_t> axes = slice->GetAxes();
+        for (size_t i = 0; i < inputShape.ranks; i++) {
+            beginVec.push_back(0);
+            endVec.push_back(inputShape.dims[i]);
+        }
+
+        if (axes.empty()) {
+            for (size_t i = 0; i < inputShape.ranks; i++) {
+                beginVec[i] = starts[i];
+                if (sizes[i] == -1) {
+                    continue;
+                }
+                endVec[i] = starts[i] + sizes[i];
+                if (beginVec[i] < 0 && endVec[i] >= 0)
```

Comment:
```
input dimensions =[3, 4, 5]
begin[0] = starts[0] = -1
sizes[0] = 2
```
expected  end[0] = -2, but the code will output end[0] = 3,

Maybe you need to do like below
```
// slice one axis
   begin.push_back(starts[i]);
   if (sizes[i] = -1) {
       end.push_back(starts[i] < 0 ? (-inputShape.dims[i]) : inputShape.dims[i]);
   } else {
       end.push_back(starts[i] < 0 ? (starts[i] - sizes[i] - 1) : (starts[i] + sizes[i]));
   }
```

Reply:
I think I misunderstanding the meaning of the `size` when the `start` is a negative value. I will try your implementation.

As for the failed case. 
when the 
```
input dimensions =[3, 4, 5]
begin[0] = starts[0] = -2
sizes[0] = 2
```

The start will be -2, and the end will be -3. In this situation, the OpenVINO op will meet an error. Because the index of the end is lower the index of the start. (Where -3 means the the dimensions[0], -2 means the dimension[1]). The OpenVINO op asks for the end behind the start.

Reply:
>expected end[0] = -2, but the code will output end[0] = 3

The expected end[0] should be -3. Because the `StrideSlice` would keep the index in the `begin`, and abandon the index in the `end`.

Reply:
> As for the failed case. 
> when the 
> ```
> input dimensions =[3, 4, 5]
> begin[0] = starts[0] = -2
> sizes[0] = 2
> ```
> 
> The start will be -2, and the end will be -3. In this situation, the OpenVINO op will meet an error. Because the index of the end is lower the index of the start. (Where -3 means the the dimensions[0], -2 means the dimension[1]). The OpenVINO op asks for the end behind the start.

The end[0] should be -1, so below code can get the expected shape?

```
// slice one axis
   begin.push_back(starts[i]);
   if (sizes[i] = -1) {
       end.push_back(starts[i] < 0 ? -1 : inputShape.dims[i];
   } else {
       end.push_back(starts[i] < 0 ? (starts[i] + sizes[i] - 1) : (starts[i] + sizes[i]));
   }
```

Reply:
Solved.